ballerina/tests/test.bal

```diff
@@ -0,0 +1,316 @@
+    Permissions response = check jira->/rest/api/'3/mypermissions(queries = queryParams);
+
+    boolean havePermission = response?.permissions["BROWSE_PROJECTS"]?.havePermission ?: false;
+    test:assertTrue(havePermission, "Do not have permission to browse projects");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testCreateRole() returns error? {
+
+    // Call the `post rest/api/'3/role` endpoint with the payload
+    ProjectRole response = check jira->/rest/api/'3/role.post(
+        payload = {
+            name: "Developer",
+            description: "Role for developers"
+        }
+    );
+
+    // Assertions
+    test:assertEquals(response.name, "Developer", msg = "Role name does not match.");
+    test:assertEquals(response.description, "Role for developers", msg = "Role description does not match.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testDeleteRole() returns error? {
+    // Define the role ID to delete
+    int roleId = 10009; 
+    http:Response response = check jira->/rest/api/'3/role/[roleId].delete();
+
+    // Assertions to verify that deletion was successful
+    test:assertEquals(response.statusCode, 204, msg = "Expected 204 No Content status for successful deletion.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testCreateProject() returns error? {
+
+    // Call the `createProject` endpoint.
+    ProjectIdentifiers response = check jira->/rest/api/'3/project.post(
+        payload={
+        key: "EX",
+        name: "Example",
+        projectTypeKey: "business",
+        leadAccountId:"712020:5d71be5f-debd-474f-9ec9-a97b9023ea4e"
+        }
+    );
+    test:assertNotEquals(response.id,null, msg = "Unsuccessful project creation");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testGetProjectSecurityLevelLive() returns error? {
+    string projectKeyOrId = "EX";  // Specify a valid project key or ID here
+
+    // Call the live API to retrieve the security level details for the project.
+    ProjectIssueSecurityLevels response = check jira->/rest/api/'3/project/[projectKeyOrId]/securitylevel();
+
+    
+
+    // Assertions to verify expected properties in the live response
+    test:assertNotEquals(response.levels.length(), 0, msg = "No security levels found for project.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testGetApplicationRoles() returns error? {
+    ApplicationRole[] response = check jira->/rest/api/'3/applicationrole();
+
+    // Validate the array is not empty and has expected properties in each item.
+    test:assertNotEquals(response.length(), 0, msg = "Expected non-empty application roles array");
+
+    foreach var role in response {
+        test:assertNotEquals(role.name, "", msg = "Application role name should not be empty");
+        test:assertNotEquals(role.key, "", msg = "Application role key should not be empty");
```

Comment:
Rather than checking if the `role.name, ""` let's check the actual string value.

Reply:
Applicable for other places as well.

Reply:
I have used a foreach there, So the role.name and role.key changes, Thats why I did not check for actual string.

Reply:
But you have only one record in that array, right? So there will be only one `role.name` and `role.key`. You can assert the values.